### PR TITLE
fix: DEV-3754: Memory usage of repeater initialization causing crashes

### PR DESCRIPTION
**Pull request by bmartel**

### CODE REVIEW

File: src/sdk/lsf-sdk.js [hunk 1]
```diff
@@ -184,7 +184,7 @@ export class LSFWrapper {
     try {
       const LSF = await resolveLabelStudio();
 
-      this.globalLSF = window.LabelStudio === LSF;
+      window.LabelStudio === LSF;
```
Comment (nicholasrq) [inline src/sdk/lsf-sdk.js:187]: not sure what you are trying to achieve here

**Review by nicholasrq — APPROVED** [review]

File: src/sdk/lsf-sdk.js [hunk 2]
```diff
@@ -384,8 +390,7 @@ export class LSFWrapper {
         c.history.safeUnfreeze();
       }
     }
-
-    const first = this.annotations[0];
+    const first = this.annotations?.length ? this.annotations[0] : null;
```
Comment (hlomzik) [inline src/sdk/lsf-sdk.js:393]: So it basically changes `undefined` to `null`? Is it required? Or `annotations` can be not defined? `?.[0]` will do the trick

File: src/sdk/lsf-sdk.js (hunk 1 above)
Comment (bmartel) [reply src/sdk/lsf-sdk.js:187]: It was an old idea leftover from a different take on this whole thing, I had changed everything to use loose references and was removing any strictly unused references like this one. Honestly its probably not doing anything given how this evolved.

Comment (hlomzik) [reply src/sdk/lsf-sdk.js:187]: Unused global var?

**Review by hlomzik — CHANGES_REQUESTED** [review]

File: src/sdk/lsf-sdk.js (hunk 1 above)
Comment (bmartel) [reply src/sdk/lsf-sdk.js:187]: search all the codebases and there is no reference to `globalLSF`. Its just another variable we had set.

File: src/sdk/lsf-sdk.js (hunk 2 above)
Comment (bmartel) [reply src/sdk/lsf-sdk.js:393]: there were warnings all over about checking the index directly, index `0` was out of bounds. It was repeatedly doing this many times. The warnings all state to check length first. Something about mobx observable arrays does not like strictly checking the value directly like that.

Comment (bmartel) [reply src/sdk/lsf-sdk.js:393]: Its the way mobx patches observables, its doing it in a way that the object will always exist, but checking indices on the underlying array can be unsafe.

Comment (bmartel) [reply src/sdk/lsf-sdk.js:393]: Screen Shot 2022-11-22 at 11 49 39 AM
This is what happens without the change I made, and it does this the same if you try and check with the optional `?.[0]`

Comment (bmartel) [reply src/sdk/lsf-sdk.js:393]: Screen Shot 2022-11-22 at 11 52 35 AM
and here is the follow through on the offending line, you can see I tried this exactly

File: src/sdk/lsf-sdk.js (hunk 1 above)
Comment (bmartel) [reply src/sdk/lsf-sdk.js:187]: Just saw this now, its supposed to have been entirely removed not just the assignement 🤣

File: src/sdk/lsf-sdk.js (hunk 2 above)
Comment (hlomzik) [reply src/sdk/lsf-sdk.js:392]: oh well... and that's not a thing you'll comment about on every occurence...
that's cool you fixed this at least in these places

**Review by hlomzik — APPROVED** [review]

**Pull request merged**